### PR TITLE
[stdlib] implement _copyContents for UnsafeRawBufferPointer

### PR DESCRIPTION
**Pull request by glessard**
Implements `_copyContents` for `Unsafe[Mutable]RawBufferPointer`, which enables a fast path when filling up `Array` or `UnsafeMutableBufferPointer<UInt8>`.

Resolves SR-9604 (rdar://52529574).

### CODE REVIEW

**Comment by glessard** [comment]
Waiting on https://github.com/apple/swift/pull/38827 for benchmarking.

**Comment by glessard** [comment]
@swift-ci please benchmark

**Comment by glessard** [comment]
@swift-ci please smoke test

File: test/stdlib/UnsafeRawBufferPointer.swift [hunk 1]
```diff
@@ -407,6 +407,18 @@ UnsafeRawBufferPointerTestSuite.test("subscript.range.wide") {
   buffer[0..<2] = buffer[0..<3]
 }
 
+UnsafeRawBufferPointerTestSuite.test("_copyContents") {
+  let a = Array<UInt8>(0..<20)
+  let b = UnsafeMutableBufferPointer<UInt8>.allocate(capacity: 10*a.count)
+  defer { b.deallocate() }
+  var (unwritten, written) = a.withUnsafeBytes {
+    bytes in
+    bytes._copyContents(initializing: b)
```
Comment (atrick) [inline test/stdlib/UnsafeRawBufferPointer.swift:416]: Out of curiosity, does this test behave any differently before and after your change?

**Review by atrick — APPROVED** [review]
Wow, that's a pretty big missing performance feature.

File: test/stdlib/UnsafeRawBufferPointer.swift (hunk 1 above)
Comment (glessard) [reply test/stdlib/UnsafeRawBufferPointer.swift:416]: The test itself behaves the same, but it runs faster. I wanted to make sure the new code path was explicitly used somewhere in the tests.

Comment (atrick) [reply test/stdlib/UnsafeRawBufferPointer.swift:416]: You could make a microbenchmark out of the original bug and add it to swift/benchmarks if you're so inclined.

Comment (glessard) [reply test/stdlib/UnsafeRawBufferPointer.swift:416]: That's done: see "RawBuffer.copyContents" above.

**Pull request merged**